ote/src/clj/ote/services/transport.clj

```diff
@@ -111,11 +111,19 @@
 (defn delete-transport-operator!
   "Delete transport operator by id"
   [nap-config db user id]
-    (authorization/with-transport-operator-check
-      db user id
-      #(do
-         (operators/delete-transport-operator db {:operator-group-name (str "transport-operator-" id)})
-         id)))
+  (let [operator-services (specql/fetch db
+                                        ::t-service/transport-service
+                                        #{::t-service/id}
+                                        {:t-service/transport-operator-id id})]
+    ;; delete only if operator-services = nil
+    (if (nil? operator-services)
+      (authorization/with-transport-operator-check
+        db user id
+        #(do
+           (operators/delete-transport-operator db {:operator-group-name (str "transport-operator-" id)})
+           id))
+    {:status 401
```

Comment:
If we're nitpicking, response should be 403 .
401 means it's an authentication problem and would succeed with proper authentication headers.  403 means request itself was valid and authenticated but server rejected it due to some logic reason

401: https://tools.ietf.org/html/rfc7235#section-3.1
403: https://tools.ietf.org/html/rfc2616#section-10.4.4

Reply:
Agreed. I thought about this, but somehow I decided that 401 close enough :)